backend/internal/controllers/middleware/requirePermission.go

```diff
@@ -0,0 +1,56 @@
+package middleware
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/TritonSE/words-alive/internal/database"
+	"github.com/TritonSE/words-alive/internal/models"
+)
+
+// HTTP middleware that validates the Authorization header against the provided authenticator
+func RequirePermission(adminDB database.AdminDatabase, p models.Permission) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
```

Comment:
Let's actually do the check that the enum is valid here, before the return statement. If it fails, go ahead and just `panic()`. That way if we've misconfigured any of the route handlers the backend won't even start, making it more likely we catch this in dev.